### PR TITLE
[Proposal][WIP][TwigBundle] Extract error handling into separate ErrorPageBundle

### PR DESCRIPTION
**Pull request by sustmi**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no?
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

- [ ] update `CHANGELOG.md` files
- [ ] update `UPGRADE.md` files
- [ ] update symfony/symfony-docs
- [ ] fix `composer.json` in ErrorPageBundle (it was copy-pasted from TwigBundle)
- [ ] fix `composer.json` in TwigBundle (it may contain dependencies that are no longer needed)
- [ ] create new `symfony/error-page-bundle` package on Packagist
- [ ] update `symfony/symfony-standard` to use the new bundle
- [ ] add backward compatibility with deprecations?

I suggest moving error handling (ie. exception listener service, exception controller, error templates and error preview controller) from TwigBundle to new ErrorPageBundle.

Error handling was moved to TwigBundle from FrameworkBundle back in Symfony 2.0 (see https://github.com/symfony/symfony/commit/3749ad43f48fbe718a1999ecd3e54d7803e4f11d) but in my opinion it does not belong there.

This change is BC-breaking so I guess it can be done only in next major release (Symfony 4.0).
What is the correct way to do such BC breaks. For example should there be a deprecated `twig.exception_listener.controller` service that is an alias pointing to the new `error_page.exception_listener.controller` service?
And how should I specify that 

What do you think about this PR?

### CODE REVIEW

**Comment by sstok** [comment]
This bundle still requires the TwigBundle to able to render the template, so why should this be extracted to a separate bundle?

**Comment by sustmi** [comment]
@sstok I think that the error page functionality simply does not belong into TwigBundle. If we separate the bundles, TwigBundle will remain with just one responsibility - to "glue" Twig into Symfony Framework. Similarly ErrorPageBundle will have one responsibility - to bring fancy customizable error pages into Symfony Framework.
Right now, TwigBundle has both of these responsibilities. I can imagine that someone wanted to use TwigBundle without the exception handling.

> This bundle still requires the TwigBundle to able to render the template, so why should this be extracted to a separate bundle?
In my opinion, "X requires Y" relation does not imply "Y contains X" relation.

**Comment by fabpot** [comment]
I don't see the point. What value does it bring?

**Comment by sustmi** [comment]
Recently, PR https://github.com/symfony/symfony/pull/20951 that redesigned the exception page was merged. I noticed that the new design looks different from the exception page that Debug component provides. Also, the highlighted code in TwigBundle's error page uses different colour schema than for example WebProfilerBundle's `_profiler_open_file` action. Lastly, the debug output from VarDumper component uses another (dark) colour schema. I thought that it would be nice to make them all share the same design.
However, it took me a while to figure out what is part of the *ErrorPage* functionality (eg. which resources are relevant) and what is the other stuff in TwigBundle. So I thought that I could save the time for other people in future by extracting it to a separate bundle to make it clearer what is what.

**TL;DR:** The point is less confusing code. However I would understand if the necessary changes and BC breaks were not worth it. :(

**Comment by nicolas-grekas** [comment]
I don't see the need also, I'd be -1 here

**Comment by stof** [comment]
The reason why the WebProfilerBundle duplicates the exception rendering instead of reusing the TwigBundle one directly is because we want to make the WebProfilerBundle UI available for Silex too for instance.

**Comment by sustmi** [comment]
Ok, I respect your opinions.
Just to clarify my intention: I wanted to make some component on which both ErrorPageBundle and WebProfilerBundle (and maybe DebugBundle) would be dependent. This new component would provide ways to implement the unified look.
The first step in doing this was to understand how all the different error page renderers work.
I had difficulties to understand which parts of TwigBundle are responsible for error pages and which not. Now that I discovered the details I thought that it may be nice of me to make it clear for others too by refactoring the code.
But I get that if the error pages generation code is touched very rarely it may not be worth refactoring it.
